pkg/promclient/promclient.go

```diff
@@ -242,24 +244,53 @@ func Snapshot(ctx context.Context, logger log.Logger, base *url.URL, skipHead bo
 	return path.Join("snapshots", d.Data.Name), nil
 }
 
+type ThanosQueryParams struct {
+	Deduplicate             bool
+	PartialResponseStrategy storepb.PartialResponseStrategy
+}
+
+func (p *ThanosQueryParams) AddTo(values url.Values) error {
+	values.Add("dedup", fmt.Sprintf("%v", p.Deduplicate))
+
+	var partialResponseValue string
+	switch p.PartialResponseStrategy {
+	case storepb.PartialResponseStrategy_WARN:
+		partialResponseValue = strconv.FormatBool(true)
+	case storepb.PartialResponseStrategy_ABORT:
+		partialResponseValue = strconv.FormatBool(false)
+	default:
+		return errors.Errorf("unknown partial response strategy %v", p.PartialResponseStrategy)
+	}
+
+	// TODO(bwplotka): Apply change from bool to strategy in Query API as well.
+	values.Add("partial_response", partialResponseValue)
+
+	return nil
+}
+
 // QueryInstant performs instant query and returns results in model.Vector type.
-func QueryInstant(ctx context.Context, logger log.Logger, base *url.URL, query string, t time.Time, dedup bool) (model.Vector, error) {
+func QueryInstant(ctx context.Context, logger log.Logger, base *url.URL, query string, t time.Time, extra ThanosQueryParams) (model.Vector, []string, error) {
```

Comment:
maybe `t time.Time` and `query` should go into `ThanosQueryParams` ?

Reply:
disagree as this function does not make any sense without that -> I think it makes sense to separate custom "Thanos" params, what do you think?

Reply:
Maybe `ThanosQueryParams` should be `QueryOptions` and then it's all good?